### PR TITLE
Make `Stats::frameRead` and `Stats::frameWritten` accept `FrameType` …

### PR DESCRIPTION
**Pull request by th0br0**
…instead of `std::string`

@lexs

### CODE REVIEW

File: src/Stats.h [hunk 1]
```diff
@@ -27,8 +27,8 @@ class Stats {
 
   virtual void bytesWritten(size_t bytes) = 0;
   virtual void bytesRead(size_t bytes) = 0;
-  virtual void frameWritten(const std::string& frameType) = 0;
-  virtual void frameRead(const std::string& frameType) = 0;
+  virtual void frameWritten(const FrameType frameType) = 0;
```
Comment (lexs) [inline src/Stats.h]: you can kill const here as it's a value type now

File: src/ConnectionAutomaton.cpp [hunk 2]
```diff
@@ -274,10 +274,7 @@ void ConnectionAutomaton::resumeStreams() {
 void ConnectionAutomaton::processFrame(std::unique_ptr<folly::IOBuf> frame) {
   auto frameType = FrameHeader::peekType(*frame);
```
Comment (lexs) [inline src/ConnectionAutomaton.cpp]: very nit: i'd kill this whitespace

**Comment by lexs** [comment]
LGTM, letting @lehecka merge

**Comment by lehecka** [comment]
I am not sure if FrameType enum is part of the public API. Can you verify that?

**Comment by th0br0** [comment]
How do you mean `public api`?
It's part of the spec as of https://github.com/ReactiveSocket/reactivesocket/blob/master/Protocol.md#frame-types (although,  although looking at the protocol definition, there's a discrepancy between our definitions and the official protocol spec - `REQUEST_SUB` is missing from the latter)

**Comment by lehecka** [comment]
public API of the cpp library, not the spec.
I am not sure if the application code has access to the internal enum type.

**Comment by th0br0** [comment]
`Frame.h` is pulled in via `reactivesocket-plugins`.

**Pull request closed without merging**